src/SFA.DAS.EmployerApprenticeshipsService.Application/Commands/RenameEmployerAccount/RenameEmployerAccountCommandHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.EAS.Application.Validation;
+using SFA.DAS.EAS.Domain.Data;
+using SFA.DAS.EAS.Domain.Interfaces;
+
+namespace SFA.DAS.EAS.Application.Commands.RenameEmployerAccount
+{
+    public class RenameEmployerAccountCommandHandler : AsyncRequestHandler<RenameEmployerAccountCommand>
+    {
+        private readonly IEmployerAccountRepository _accountRepository;
+        private readonly IValidator<RenameEmployerAccountCommand> _validator;
+        private readonly IHashingService _hashingService;
+
+        public RenameEmployerAccountCommandHandler(IEmployerAccountRepository accountRepository, IValidator<RenameEmployerAccountCommand> validator, IHashingService hashingService)
+        {
+            _accountRepository = accountRepository;
+            _validator = validator;
+            _hashingService = hashingService;
+        }
+
+        protected override async Task HandleCore(RenameEmployerAccountCommand message)
+        {
+            var validationResult = _validator.Validate(message);
```

Comment:
Might be worth in the future changing this to ValidateAsync and validating as to wether the user is an Owner